src/components/picker/helpers/usePickerSearch.tsx

```diff
@@ -3,33 +3,35 @@ import _ from 'lodash';
 import {PickerProps} from '../types';
 import {getItemLabel as getItemLabelPresenter, shouldFilterOut} from '../PickerPresenter';
 
-type UsePickerSearchProps = Pick<PickerProps, 'showSearch' | 'onSearchChange' | 'children' | 'getItemLabel'>;
+type UsePickerSearchProps = Pick<PickerProps, 'showSearch' | 'onSearchChange' | 'children' | 'getItemLabel' | 'items'>;
 
 const usePickerSearch = (props: UsePickerSearchProps) => {
-  const {showSearch, onSearchChange, children, getItemLabel} = props;
+  const {showSearch, onSearchChange, children, getItemLabel: getItemLabelPickerProp, items} = props;
   const [searchValue, setSearchValue] = useState('');
 
-  const filteredChildren = useMemo(() => {
+  const filterItems = useCallback((items: any) => {
     if (showSearch && !_.isEmpty(searchValue)) {
-      // @ts-expect-error need to fix children type
-      return _.filter(children, child => {
-        // @ts-expect-error need to fix children type to be based on PickerItemProps
-        const {label, value, getItemLabel: childGetItemLabel} = child.props;
-        const itemLabel = getItemLabelPresenter(label, value, childGetItemLabel || getItemLabel);
+      return _.filter(items, item => {
+        const {label, value, getItemLabel} = item.props || item;
+        const itemLabel = getItemLabelPresenter(label, value, getItemLabel || getItemLabelPickerProp);
```

Comment:
I think you've reversed the order (I think it should be `getItemLabelPickerProp || getItemLabel`)
I'd also consider keeping the old naming convention:
```
const {label, value, getItemLabel: childGetItemLabel} = item.props || item;
const itemLabel = getItemLabelPresenter(label, value, childGetItemLabel || getItemLabel);
```
This way only the local prop is renamed, which does not affect the whole file.

Reply:
I’ve suggested returning the `getItemLabelPickerProp` to the original `getItemLabel`, not sure if you’ve missed it or it was on purpose